src/main/java/io/vertx/ext/hawkular/impl/InventoryReporter.java

```diff
@@ -120,72 +130,41 @@ public InventoryReporter(Vertx vertx, VertxHawkularOptions options, Context cont
       httpClient = vertx.createHttpClient(httpClientOptions);
       }
     );
   }
   public void report() {
     context.runOnContext(aVoid -> {
-      reportFeed().setHandler(ar1 -> {
-        if (ar1.succeeded()) {
-          createVertxRootResourceType().setHandler(ar2 -> {
-            if (ar2.succeeded()) {
-              createVertxRootResource().setHandler(ar3 -> {
-                if (ar3.succeeded()) {
-                  System.out.println("Done with vertx resource");
-                  createEventbusResourceType().setHandler(ar4 -> {
-                    if (ar4.succeeded()) {
-                      createEventbusResource().setHandler(ar5 -> {
-                        if (ar5.succeeded()) {
-                          createGaugeMetricType().setHandler(ar6 -> {
-                            if (ar6.succeeded()) {
-                              createEventbusHandlerMetric().setHandler(ar7 -> {
-                                if (ar7.succeeded()) {
-                                  associateGaugeMetricTypeWithEventbusResourceType().setHandler(ar8 -> {
-                                    if (ar8.succeeded()) {
-                                      System.out.println("Done with event bus handler");
-                                    } else {
-                                      System.err.println(ar8.cause().getLocalizedMessage());
-                                    }
-                                  });
-                                } else {
-                                  System.err.println(ar7.cause().getLocalizedMessage());
-                                }
-                              });
-                            } else {
-                              System.err.println(ar6.cause().getLocalizedMessage());
-                            }
-                          });
-                        } else {
-                          System.err.println(ar5.cause().getLocalizedMessage());
-                        }
-                      });
-                    } else {
-                      System.err.println(ar4.cause().getLocalizedMessage());
-                    }
-                  });
-                } else {
-                  System.err.println(ar3.cause().getLocalizedMessage());
-                }
-              });
-            } else {
-              System.err.println(ar2.cause().getLocalizedMessage());
-            }
-          });
-        } else {
-          System.err.println(ar1.cause().getLocalizedMessage());
-        }
+      Future<Void> fut1 = Future.future();
+      Future<Void> fut2 = Future.future();
+
+      reportFeed(fut1);
+      fut1.compose(aVoid1 -> {
+        reportRootResource(fut2);
+      }, fut2);
+      fut2.setHandler(aVoid1 -> {
+        Future<Void> rfut1 = Future.future();
+        Future<Void> rfut2 = Future.future();
+        reportEventbusResource(rfut1);
+        reportHttpServerResource(new SocketAddressImpl(8080, "0.0.0.0"), rfut2);
```

Comment:
Next step is to detect when http servers are created (use the callbacks from the Metrics SPI).